tidb-binlog/driver/reader/kafka_consumer/kafka_go.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka_consumer
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/segmentio/kafka-go"
+	"go.uber.org/zap"
+)
+
+const (
+	kafkaGOType = "kafka-go"
+)
+
+type KafkaGO struct {
+	// client is high level api, only can consumer from offset
+	client *kafka.Reader
+	// conn is low level api, which has createTopic DeleteTopic and other more function than *kafka.Reader
+	conn *kafka.Conn
+
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+// NewKafkaGoConsumer return kafka-go consumer on specify topic and partition
+func NewKafkaGoConsumer(cfg *KafkaConfig) (Consumer, error) {
+	if len(cfg.Addr) == 0 {
+		return nil, errors.New("no available kafka address")
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	conn, err := kafka.DialLeader(ctx, "tcp", cfg.Addr[0], cfg.Topic, int(cfg.Partition))
```

Comment:
According to comment
```
// The address given to the DialContext method may not be the one that the
// connection will end up being established to, because the dialer will lookup
// the partition leader for the topic and return a connection to that server.
// The original address is only used as a mechanism to discover the
// configuration of the kafka cluster that we're connecting to.
```
this `cfg.Addr[0]` only use to get partition leader from cluster then build wrapped kafka connection to partition leader
if `cfg.Addr[0]` not work, I can add retry logic here, try use other broker address.
if partition leader is down, kafka will elect new leader, this connection need reconnect, I'll fix it.